autonomy/chain/mint.py

```diff
@@ -132,6 +132,19 @@ def get_contract(public_id: PublicId) -> Contract:
     return Contract.from_dir(directory=contract_dir)
 
 
+def transact(ledger_api: LedgerApi, crypto: Crypto, tx: Dict) -> Dict:
+    """Make a transaction and return a receipt"""
+
+    signed_tx = ledger_api.api.eth.account.sign_transaction(
+        tx,
+        private_key=crypto.private_key,
+    )
+    ledger_api.api.eth.send_raw_transaction(signed_tx.rawTransaction)
+    tx_hash = ledger_api.api.toHex(ledger_api.api.keccak(signed_tx.rawTransaction))
+
+    return ledger_api.api.eth.wait_for_transaction_receipt(tx_hash)
```

Comment:
again check out the plugin,. We have that for a reason so it's easy to move to other ledgers.